File: server-kotlin-spring/src/main/kotlin/org/openapitools/model/Connection.kt
package org.openapitools.model

import java.util.Objects
import com.fasterxml.jackson.annotation.JsonCreator
import com.fasterxml.jackson.annotation.JsonProperty
import com.fasterxml.jackson.annotation.JsonValue
import org.openapitools.model.Account
import org.openapitools.model.ConnectedFriend
import javax.validation.constraints.DecimalMax
import javax.validation.constraints.DecimalMin
import javax.validation.constraints.Email
import javax.validation.constraints.Max
import javax.validation.constraints.Min
import javax.validation.constraints.NotNull
import javax.validation.constraints.Pattern
import javax.validation.constraints.Size
import javax.validation.Valid
import io.swagger.v3.oas.annotations.media.Schema

/**
 * 
 * @param id 
 * @param created 
 * @param updated 
 * @param deleted 
 * @param objectType 
 * @param legacyId 
 * @param searchTags 
 * @param active 
 * @param valid 
 * @param account 
 * @param connection 
 * @param pendingConnection 
 * @param friend 
 * @param friendRequested 
 * @param friendRequestPending 
 * @param following 
 * @param follower 
 * @param trusted 
 * @param blocked 
 * @param friendRequestDate 
 * @param friendResponseDate 
 * @param display 
 * @param connectionViewedDate 
 * @param connectionType 
 * @param title 
 * @param pendingConnectionId 
 * @param latitude 
 * @param longitude 
 */
data class Connection(

    @Schema(example = "null", description = "")
    @get:JsonProperty("id") val id: kotlin.Long? = null,

    @Schema(example = "null", description = "")
    @get:JsonProperty("created") val created: java.time.OffsetDateTime? = null,

    @Schema(example = "null", description = "")
    @get:JsonProperty("updated") val updated: java.time.OffsetDateTime? = null,

    @Schema(example = "null", description = "")
    @get:JsonProperty("deleted") val deleted: java.time.OffsetDateTime? = null,

    @Schema(example = "null", description = "")
    @get:JsonProperty("objectType") val objectType: kotlin.String? = null,

    @Schema(example = "null", description = "")
    @get:JsonProperty("legacyId") val legacyId: kotlin.String? = null,

    @Schema(example = "null", description = "")
    @get:JsonProperty("searchTags") val searchTags: kotlin.String? = null,

    @Schema(example = "null", description = "")
    @get:JsonProperty("active") val active: kotlin.Boolean? = null,

    @Schema(example = "null", description = "")
    @get:JsonProperty("valid") val valid: kotlin.Boolean? = null,

    @field:Valid
    @Schema(example = "null", description = "")
    @get:JsonProperty("account") val account: Account? = null,

    @field:Valid
    @Schema(example = "null", description = "")
    @get:JsonProperty("connection") val connection: Account? = null,

    @field:Valid
    @Schema(example = "null", description = "")
    @get:JsonProperty("pendingConnection") val pendingConnection: ConnectedFriend? = null,

    @Schema(example = "null", description = "")
    @get:JsonProperty("friend") val friend: kotlin.Boolean? = null,

    @Schema(example = "null", description = "")
    @get:JsonProperty("friendRequested") val friendRequested: kotlin.Boolean? = null,

    @Schema(example = "null", description = "")
    @get:JsonProperty("friendRequestPending") val friendRequestPending: kotlin.Boolean? = null,

    @Schema(example = "null", description = "")
    @get:JsonProperty("following") val following: kotlin.Boolean? = null,

    @Schema(example = "null", description = "")
    @get:JsonProperty("follower") val follower: kotlin.Boolean? = null,

    @Schema(example = "null", description = "")
    @get:JsonProperty("trusted") val trusted: kotlin.Boolean? = null,

    @Schema(example = "null", description = "")
    @get:JsonProperty("blocked") val blocked: kotlin.Boolean? = null,

    @Schema(example = "null", description = "")
    @get:JsonProperty("friendRequestDate") val friendRequestDate: java.time.OffsetDateTime? = null,

    @Schema(example = "null", description = "")
    @get:JsonProperty("friendResponseDate") val friendResponseDate: java.time.OffsetDateTime? = null,

    @Schema(example = "null", description = "")
    @get:JsonProperty("display") val display: kotlin.String? = null,

    @Schema(example = "null", description = "")
    @get:JsonProperty("connectionViewedDate") val connectionViewedDate: java.time.OffsetDateTime? = null,

    @Schema(example = "null", description = "")
    @get:JsonProperty("connectionType") val connectionType: Connection.ConnectionType? = null,

    @Schema(example = "null", description = "")
    @get:JsonProperty("title") val title: kotlin.String? = null,

    @Schema(example = "null", description = "")
    @get:JsonProperty("pendingConnection_id") val pendingConnectionId: kotlin.Long? = null,

    @Schema(example = "null", description = "")
    @get:JsonProperty("latitude") val latitude: kotlin.Double? = null,

    @Schema(example = "null", description = "")
    @get:JsonProperty("longitude") val longitude: kotlin.Double? = null
) {

    /**
    * 
    * Values: PRIMARY,SECONDARY,DEPENDENT,ACQUAINTANCE
    */
    enum class ConnectionType(@get:JsonValue val value: kotlin.String) {

        PRIMARY("PRIMARY"),
        SECONDARY("SECONDARY"),
        DEPENDENT("DEPENDENT"),
        ACQUAINTANCE("ACQUAINTANCE");

        companion object {
            @JvmStatic
            @JsonCreator
            fun forValue(value: kotlin.String): ConnectionType {
                return values().firstOrNull{it -> it.value == value}
                    ?: throw IllegalArgumentException("Unexpected value '$value' for enum 'Connection'")
            }
        }
    }

}

File: server-aspnet-fastendpoints/src/Org.OpenAPITools/Models/Connection.cs
namespace Org.OpenAPITools.Models;


/// <summary>
/// 
/// </summary>
public class Connection 
{
    public long Id { get; set; }
    public DateTime Created { get; set; }
    public DateTime Updated { get; set; }
    public DateTime Deleted { get; set; }
    public string ObjectType { get; set; }
    public string LegacyId { get; set; }
    public string SearchTags { get; set; }
    public bool Active { get; set; }
    public bool Valid { get; set; }
    public Account Account { get; set; }
    public Account VarConnection { get; set; }
    public ConnectedFriend PendingConnection { get; set; }
    public bool Friend { get; set; }
    public bool FriendRequested { get; set; }
    public bool FriendRequestPending { get; set; }
    public bool Following { get; set; }
    public bool Follower { get; set; }
    public bool Trusted { get; set; }
    public bool Blocked { get; set; }
    public DateTime FriendRequestDate { get; set; }
    public DateTime FriendResponseDate { get; set; }
    public string Display { get; set; }
    public DateTime ConnectionViewedDate { get; set; }
    
    /// <summary>
    /// Gets or Sets ConnectionType
    /// </summary>
    public enum ConnectionTypeEnum
    {
        
        /// <summary>
        /// Enum PRIMARYEnum for PRIMARY
        /// </summary>
        PRIMARYEnum = 1,
        
        /// <summary>
        /// Enum SECONDARYEnum for SECONDARY
        /// </summary>
        SECONDARYEnum = 2,
        
        /// <summary>
        /// Enum DEPENDENTEnum for DEPENDENT
        /// </summary>
        DEPENDENTEnum = 3,
        
        /// <summary>
        /// Enum ACQUAINTANCEEnum for ACQUAINTANCE
        /// </summary>
        ACQUAINTANCEEnum = 4
    }

    public ConnectionTypeEnum ConnectionType { get; set; }
    public string Title { get; set; }
    public long PendingConnectionId { get; set; }
    public double Latitude { get; set; }
    public double Longitude { get; set; }
}


